src/index.ts

```diff
@@ -84,13 +87,28 @@ type IntersectionOfReturns<Funcs extends AnyFunction[]> = UnionToIntersection<
   ReturnType<Funcs[number]>
 >;
 
+type ApproveTokenFunctionsKeys = keyof ApproveTokenFunctions<any>;
+
+type PartialSDKResult<
+  Config extends ConstructBaseInput,
+  Funcs extends [SDKFunction<Config>, ...SDKFunction<Config>[]]
+> = IntersectionOfReturns<Funcs> extends ApproveTokenFunctions<any> // if there are ApproveTokenFunctions in the intersection
+  ? // which means constructApproveToken was passed in Funcs
+    Omit<IntersectionOfReturns<Funcs>, ApproveTokenFunctionsKeys> &
+      ApproveTokenFunctions<
+        // infer what TxResponse was used in Config
+        Config extends SDKConfig<infer TxResponse> ? TxResponse : unknown
+        // and make the ApproveTokenFunctions<unknow> in the intersection a specific ApproveTokenFunctions<TxResponse>
+      >
+  : IntersectionOfReturns<Funcs>;
```

Comment:
omg that was dense 
But hopefully levelled up my TS, and we have nice inference of approval func return 👍